site-service/src/test/java/uk/ac/ox/ndph/mts/site_service/SiteServiceIntegrationTests.java

```diff
@@ -0,0 +1,88 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import uk.ac.ox.ndph.mts.site_service.exception.RestException;
+import uk.ac.ox.ndph.mts.site_service.repository.FhirRepository;
+
+@SpringBootTest(properties = { "server.error.include-message=always", "spring.main.allow-bean-definition-overriding=true" })
+@ActiveProfiles("test-all-required")
+@AutoConfigureMockMvc
+class SiteServiceIntegrationTests {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    public FhirRepository repository;
+
+    @Test
+    void TestPostSite_WhenValidInput_Returns201AndId() throws Exception {
+        // Arrange
+        when(repository.saveOrganization(any(Organization.class))).thenReturn("123");
+        when(repository.saveResearchStudy(any(ResearchStudy.class))).thenReturn(any(ResearchStudy.class));
+        
+        String jsonString = "{\"name\": \"name\", \"alias\": \"alias\"}";
+        // Act + Assert
+        this.mockMvc
+                .perform(post("/site").contentType(MediaType.APPLICATION_JSON).content(jsonString))
+                .andDo(print()).andExpect(status().isCreated()).andExpect(content().string(containsString("123")));
+    }
+
+    @Test
+    void TestPostSite_WhenInvalidInput_ReturnsUnprocessableEntityAndDescription() throws Exception {
+        // Arrange
+        when(repository.saveOrganization(any(Organization.class))).thenReturn("123");
+
+        String jsonString = "{\"name\": \"\", \"alias\": \"alias\"}";
+        // Act + Assert
+        var error = this.mockMvc
+                .perform(post("/site").contentType(MediaType.APPLICATION_JSON).content(jsonString))
+                .andDo(print()).andExpect(status().isUnprocessableEntity()).andReturn().getResolvedException().getMessage();
+        assertThat(error, containsString("Name"));
+    }
+
+    @Test
+    void TestPostSite_WhenValidInputAndRepositoryThrows_ReturnsBadGateway() throws Exception {
+        // Arrange
+        when(repository.saveOrganization(any(Organization.class))).thenThrow(new RestException("test error"));
+        
+        
+        String jsonString = "{\"name\": \"name\", \"alias\": \"alias\"}";
+        // Act + Assert
+        var error = this.mockMvc
+                .perform(post("/site").contentType(MediaType.APPLICATION_JSON).content(jsonString))
+                .andDo(print()).andExpect(status().isBadGateway()).andReturn().getResolvedException().getMessage();
+        assertThat(error, containsString("test error"));
+    }
+
+    @Test
+    void TestPostSite_WhenValidParentInput_Returns201AndId() throws Exception {
+        // Arrange
+        when(repository.saveOrganization(any(Organization.class))).thenReturn("123");
+        when(repository.saveResearchStudy(any(ResearchStudy.class))).thenReturn(any(ResearchStudy.class));
+
+        String jsonString = "{\"name\": \"name\", \"alias\": \"alias\", \"parentFhirId\": \"parentFhirId\"}";
+        // Act + Assert
+        this.mockMvc
```

Comment:
You should validate that the org is connect to the parent in the request

Reply:
This will be done later